### PR TITLE
#20 - check nodes on ping 

### PR DESCRIPTION
**Pull request by robertkowalski**
This is a fix for #20

### CODE REVIEW

**Comment by prdn** [comment]
I have tested with https://github.com/bitfinexcom/grenache-grape and works like a charm.
Without removing nodes we were having lookups and announces that were trying to reach failing nodes causing requests to complete only after 2.5 seconds (k-rpc timeout), slowing down the entire network.
@feross would be great to have it included in the repo.

Thanks

**Comment by robertkowalski** [comment]
build broken because Object.assign usage, should turn green if we drop 0.10 and 0.10 which are not supported anymore (PR at https://github.com/webtorrent/bittorrent-dht/pull/171)

**Comment by robertkowalski** [comment]
heya, i rebased against the latest master. 

this runs in production for our systems just fine since a few months, previously we had the issue @prdn described for long-running nodes

**Comment by DiegoRBaquero** [comment]
This look good, although not at all my area of expertise. @feross ?

File: client.js [hunk 1]
```diff
@@ -39,12 +40,19 @@ function DHT (opts) {
...
+  this._bucketCheckInterval = null
+  this._bucketOutdatedTimeSpan = opts.timeBucketOutdated || BUCKET_OUTDATED_TIMESPAN
 
   this.listening = false
   this.destroyed = false
   this.nodeId = this._rpc.id
   this.nodes = this._rpc.nodes
 
+  this.nodes.on('ping', function (nodes) {
+    self._checkNodes(nodes)
```
Comment (robertkowalski) [inline client.js]: todo: fix

**Comment by mafintosh** [comment]
Tested this and it looks good.

**Comment by mafintosh** [comment]
7.8.0

**Pull request merged**